daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/NonBaseTenTextNumberParser.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.processors.parsers
+
+import java.math.{ BigInteger => JBigInt }
+import java.lang.{ Number => JNumber }
+
+import org.apache.daffodil.dpath.NodeInfo
+import org.apache.daffodil.processors.ElementRuntimeData
+
+class ConvertNonBaseTenTextNumberParser(
+  override val context: ElementRuntimeData,
+  base: Int)
+  extends TextPrimParser {
+
+  override lazy val runtimeDependencies = Vector()
+
+  private val primNumeric = context.optPrimType.get.asInstanceOf[NodeInfo.PrimType.PrimNumeric]
+
+  final def parse(state: PState): Unit = {
+
+    val node = state.simpleElement
+    val baseStr = node.dataValueAsString
+
+    if (baseStr == "") {
+      PE(state, "Unable to parse %s from empty string", context.optPrimType.get.globalQName)
+      return
+    }
+
+    // must explicitly check for and error on text that start with + or -
+    // because DFDL does not allow a leading sign character, but parseInt and
+    // friends will accept them
+    val firstChar = baseStr(0)
+    if (firstChar == '-' || firstChar == '+') {
```

Comment:
Please add to comment that DFDL spec says textNumberPattern is not used when textStandardBase is not 10.

Reply:
WIll do